pytest_girder/pytest_girder/fixtures.py

```diff
@@ -17,7 +19,7 @@ def bcrypt():
         yield hashpw
 
 
-@pytest.fixture
+@pytest.fixture(scope='module')
```

Comment:
By sharing this database fixture across modules, you are effectively sharing state between test functions.  This is something we are actively trying to avoid not just for increased parallelism, but to enhance the quality and debug-ability of our tests.

I'm actually somewhat surprised this even works with the current tests.  It makes me think that I might not actually understand what's going on.  :confused:

Reply:
I'm going to be doing my lightning talk tomorrow about this very subject, hoping it leads to some design decisions. The current implementation of the `db` fixtures is already a little problematic, but there are engineering tradeoffs to consider either way.

Function-scope fixtures may depend on module-scope fixtures, but not the other way around.

The reason I think this works is because our other fixtures clean up their documents on teardown.

Reply:
Fixtures clean up their documents, but tests do not.  I'm on the side of explicitly making all test functions independent even at the cost of execution time.  It sounds like something we can discuss after your lightning talk tomorrow.